Receptai.client/components/Banner/Banner.vue

```diff
@@ -2,46 +2,140 @@
+    if (!moveBannersInterval) {
+        moveBannersInterval = setInterval(nextSlide, 2000);
+    }
+}
+
+function nextSlide(): void {
+    const maxOffset = recipeList.value.length * -bannerComponentWidth + bannerWidth.value;
+    const nextOffset = bannerOffset.value - bannerComponentWidth;
+    bannerOffset.value = bannerOffset.value == maxOffset ? 0 : bannerOffset.value > 0 ? 0 : Math.max(maxOffset, nextOffset);
 }
 
-const recipeList = ref<Recipe[] | null>(null);
 
-try {
-    axios
-        .get(
-            `${config.public.baseURL}/api/v1/recipe/list?page=0&sortBy=dateCreated&sortAsc=false`
-        )
-        .then((res) => {
-            recipeList.value = res.data.elements.slice(0, 5);
-        });
-} catch (e) {
-    console.error("Error fetching recipes", e);
+function prevSlide(): void {
+    bannerOffset.value = Math.min(bannerOffset.value + bannerComponentWidth, 0);
 }
+
+function getEventX(event: MouseEvent | TouchEvent): number {
+    return event instanceof MouseEvent ? event.clientX : event.touches[0].clientX;
+}
+
+const bannerTransformStyle = computed(() => `translateX(${bannerOffset.value}px)`);
+
+
+onMounted(() => {
+    nextTick(() => {
+        const bannerContainer = document.querySelector('.w-full.mx-auto.overflow-hidden');
+        if (bannerContainer) {
+            bannerWidth.value = bannerContainer.clientWidth;
+        }
+    });
+});
 </script>
 
 <template>
-    <div v-if="recipeList" class="banner flex flex-wrap">
-        <RecipeBannerComponent v-for="item in recipeList" :key="item.id" :imageLink="item.previewImage"
-            :name="item.name" :about="item.shortDescription" :link="`/recipes/${item.id}`" />
-    </div>
-    <div v-else>
-        <p>Failed to load recipes.</p>
+    <div class="relative" @mousedown.prevent="startDrag" @touchstart.prevent="startDrag" @mousemove="drag"
+        @touchmove="drag" @mouseup="endDrag" @touchend="endDrag" @mouseleave="endDrag" @touchcancel="endDrag">
+        <div class="w-full mx-auto overflow-hidden rounded-lg border border-gray-800 bg-gray-50" ref="bannerContainer">
+            <div class="flex"
+                :style="{ transform: bannerTransformStyle, transition: isDragging ? 'none' : 'transform 0.5s ease' }">
+                <!-- Recipes -->
+                <div v-for="item in recipeList" :key="item.id" class="flex-shrink-0 mt-4">
+                    <RecipeBannerComponent :imageLink="item.previewImage" :name="item.name"
+                        :about="item.shortDescription" :link="`/recipes/${item.id}`" />
+                </div>
+            </div>
+        </div>
+        <!-- Navigation Arrows -->
+        <div class="absolute top-1/2 -left-10 transform -translate-y-1/2 cursor-pointer" @click="prevSlide">
+            <svg class="w-6 h-6 text-black" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+                <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M15 19l-7-7 7-7" />
+            </svg>
```

Comment:
While this does the job, it's better to define icons as svg files for better viewing. In this instance I would take a look at an already implemented library in the project: https://nuxt.com/modules/icon. Icons can be easily implemented like this: <Icon name="uil:github" color="black" />.